rfcs/text/0000-roadmaps+planning.md

```diff
@@ -0,0 +1,421 @@
+
+The Roadmap Shepherd MUST be assigned as the Shepherd for the RFC and ensure that the Project Leads (Authors) respond to comments, answer questions, and update the Roadmap RFC as required.
+
+### Iterating with Oversight
+[Iterating with Oversight]: #iterating-with-oversight
+
+Roadmaps MUST be approved by Oversight rather than DCP PM which is a minor variation from the current RFC Approval process for Governance RFC(s).
+
+The Roadmap Shepherd MUST send the draft Roadmap to Oversight a week in advance of a review meeting between the Project Leads and Oversight.
+
+The Project Leads MUST iterate with Oversight to review and revise the draft Roadmap until it meets with Oversight approval.
+
+Once approved, the Roadmap Shepherd will follow the normal RFC process for [approved RFC(s)](https://github.com/HumanCellAtlas/dcp-community/blob/master/rfcs/text/0001-rfc-process.md#approving-rfcs).
+
+### Updating the Roadmap
+
+The Roadmap MAY be updated with Project Lead consensus and Oversight approval for a number of reasons including but not limited to:
+
+* A Theme is to be added or deleted.
+* Additions or revision to Objectives change the original intent of the Theme.
+* Emerging conditions require substantive changes to the original Roadmap direction.
+
+Requests to update the Roadmap MUST be forwarded to the Roadmap Shepherd with a detailed description of the proposed changes, including the rationale for the updates, a timeline, and priority.
+
+The Roadmap Shepherd MUST add the request to the next Project Leads meeting agenda for review.
+
+The Roadmap Shepherd and Project Leads MUST review the update request and determine if there is rough consensus.
+
+If consensus is reached, the Roadmap Shepherd MUST refresh the Roadmap and send to Oversight. 
+
+Oversight MUST review the changes and approve.
+
+The Roadmap Shepherd MUST update the previously published Roadmap RFC in the *dcp-community* repo and announce its approval in a similar fashion to the original RFC approval.
+
+## Detailed Design for Planning and Execution
+
+### ZenHub for Project Management
+
+The [ZenHub](https://www.zenhub.com/) project management application has been partially adopted in the DCP community. This RFC fully commits the community to ZenHub as our *one source of truth* for project management and engineering plans.
+
+DCP SHOULD NOT duplicate the ZenHub DCP Board in other documents or spreadsheets. The Board and its reports represent the definitive list of DCP work items for a Release and their current state. 
+
+### DCP Product Backlog
+
+The DCP ZenHub board is the Product Backlog for DCP. All DCP component repos that will assign items to a DCP Release and Milestone MUST be added to the ZenHub *DCP Backlogs* workspace. The [DCP PM Team](https://github.com/orgs/HumanCellAtlas/teams/dcp-pm-team) in the Human Cell Atlas GitHub organization MUST have write access to all the repo(s) in the *DCP workspace*.
```

Comment:
Teams are free to schedule in the same milestone if they have high confidence in the estimation capabilities of their partner. It's a recommendations rather than a requirement. Teams are also free to _pull in_ Milestones. It's always acceptable to deliver work early. During the earlier _Quarterly Goals_ process there was feedback that the last links in a dependency chain often were unable to complete their work in the quarter, because earlier links exhausted the time.

Milestones indicate when work will be completed. The _In Progress_ pipeline indicates when work has started. There should be adequate decomposition to be able to demonstrate progress - an epic may start in Milestone 1 and end in Milestone 3 - but there should be issues along the way in other Milestones. Otherwise, we're apt to reach Milestone 3 and discover that three months of work has now slipped.

Milestones become more useful when teams can actually deliver their dependencies on schedule and build confidence with their partners. There's nothing preventing or discouraging teams from coordinating with each other especially with the focus on DCP as a service.

Reply:
Ok, based on what you're saying, if something is in progress but unclear on when it will be delivered, then it would not have a milestone. This line in the RFC says that all tickets `MUST` have a milestone assigned. If it's ok to not have a milestone and teams can coordinate however they'd like, then I'm less worried about the ability to choose to use milestones. This line would need to be updated to reflect that, though.

Reply:
No. All issues in a Release MUST have a Milestone to indicate when they will be complete. When the issue is started, then it moves to _In Progress_. Other stages to monitor progress are described in the RFC. If you cannot estimate an issue, then you need to review the section on the use of Spikes. Otherwise, we're potentially [polling other teams](https://www.youtube.com/watch?v=a8k1eTEw4rQ) to understand if work is complete, which is both inefficient and frustrating.

Reply:
If I'm understanding what you're saying, then I'm back to my original comment that I don't think adding assigning milestones has been adding value to the process. Polling is one choice of how to work together, but there's another option to collaboratively work together to implement and test a feature. To represent the work in Zenhub, one can utilize Zenhub dependency management, status tracking by the columns, demoing progress biweekly, and updating with comments and backlog refinement.

Reply:
Milestones are quite useful when staging dependencies and understanding when work will be delivered during a Release. We have an open question about how to hold teams *accountable* for their estimates to ensure that Milestones are valuable and high confidence.

Reply:
I'm not sure you have any way to hold teams _accountable_ in the corporate sense. We're all in institutions that operate as peers with a lot built on goodwill and shared objectives.

Reply:
RE estimation and milestones, I enjoy Steven Sinofsky's perspective as a former President of the Windows division. In [Learning by Shipping](https://medium.learningbyshipping.com/plans-3bcc6164e8b):

***When**. Timing is life. Plans collapse when management (or sales) says now and prod/eng says 6 mon. and no one is entirely honest. Key: **individuals doing work say how long. Key to a plan is leadership sets time constraint then work scales.**
**PS: If you’re coordinating plans across two orgs, then timing is the only universal language — and it is a language of commitment. Two teams can only work together if they share high integrity timing** — two functional teams or client/back end, etc.*